full-node/db/sov-schema-db/src/iterator.rs

```diff
@@ -28,19 +28,23 @@ pub trait SeekKeyEncoder<S: Schema + ?Sized>: Sized {
     fn encode_seek_key(&self) -> crate::schema::Result<Vec<u8>>;
 }
 
+// unused outside `std`
+#[cfg(feature = "std")]
 pub(crate) enum ScanDirection {
     Forward,
     Backward,
 }
 
 /// DB Iterator parameterized on [`Schema`] that seeks with [`Schema::Key`] and yields
 /// [`Schema::Key`] and [`Schema::Value`] pairs.
+#[cfg(feature = "std")]
 pub struct SchemaIterator<'a, S> {
```

Comment:
I think the ` SchemaIterator`and all its dependencies  can be moved to separate file and then only the file could be imported with `#[cfg(feature = "std")]`

Reply:
Ditto; let's centralize on the previous comment